File: app/src/main/java/gardezi/io/amiibofinder/view/AmiiboAdapter.java
package gardezi.io.amiibofinder.view;


import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import gardezi.io.amiibofinder.databinding.AmiiboListItemBinding;
import gardezi.io.amiibofinder.model.Amiibo;

public class AmiiboAdapter extends RecyclerView.Adapter {

    private List<Amiibo> mAmiibos = new ArrayList<>();

    public AmiiboAdapter() {
    }

    public void updateData(List<Amiibo> amiibos) {
        mAmiibos = amiibos;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        AmiiboListItemBinding binding = AmiiboListItemBinding.inflate(layoutInflater, parent, false);
        return new AmiiboViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        Amiibo amiibo = mAmiibos.get(position);
        ((AmiiboViewHolder) holder).bind(amiibo);
    }

    @Override
    public int getItemCount() {
        return mAmiibos.size();
    }

    class AmiiboViewHolder extends RecyclerView.ViewHolder {

        private final AmiiboListItemBinding binding;

        AmiiboViewHolder(AmiiboListItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        void bind(Amiibo amiibo) {
            binding.setAmiibo(amiibo);
            binding.executePendingBindings();
        }
    }
}

File: app/src/main/java/gardezi/io/amiibofinder/view/AmiiboAdapter.kt
package gardezi.io.amiibofinder.view


import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.content.Context
import android.support.v4.app.FragmentActivity
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import java.util.ArrayList

import gardezi.io.amiibofinder.databinding.AmiiboListItemBinding
import gardezi.io.amiibofinder.model.Amiibo
import gardezi.io.amiibofinder.viewmodel.AmiiboViewModel

class AmiiboAdapter() : RecyclerView.Adapter<AmiiboAdapter.AmiiboViewHolder>() {

    private var mAmiibos: List<Amiibo> = ArrayList()

    public fun updateData(amiibos: List<Amiibo>?) {
        if (amiibos != null && amiibos.isNotEmpty()) {
            mAmiibos = amiibos
            notifyDataSetChanged()
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AmiiboViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val binding = AmiiboListItemBinding.inflate(layoutInflater, parent, false)
        return AmiiboViewHolder(binding)
    }

    override fun onBindViewHolder(holder: AmiiboViewHolder, position: Int) {
        val amiibo = mAmiibos[position]
        holder.bind(amiibo)
    }

    override fun getItemCount(): Int {
        return mAmiibos.size
    }

    class AmiiboViewHolder(private val binding: AmiiboListItemBinding) : RecyclerView.ViewHolder(binding.root) {

        fun bind(amiibo: Amiibo) {
            binding.amiibo = amiibo
            binding.executePendingBindings()
        }
    }
}
